shnet_main.c

```diff
@@ -190,21 +192,22 @@ static int shnet_port_recv(struct shnet_port *port,
     recv->vec = recv->req.vec;
     recv->port = port;
 
-    if ((unsigned long)recv->req.vec[0].iov_base & (PAGE_SIZE -1)) {
+    nr_pages = recv->req.vec[0].iov_len / PAGE_SIZE + 1;
+    pg_offs = (unsigned long)recv->req.vec[0].iov_base & (PAGE_SIZE - 1);
+    if (pg_offs) {
         pr_err("Address %p is not aligned\n", recv->req.vec[0].iov_base);
-    	return post_cqe(port, recv->req.connection_id, recv->req.req_id,
-			SHNET_ERR_CODE_INV_ADDR);
+	nr_pages++;
```

Comment:
This increment is when address is not aligned.
With the above fix do you still have this question?